packages/amplify-cli/src/extensions/amplify-helpers/auth-notifications.ts

```diff
@@ -0,0 +1,100 @@
+import { $TSAny, $TSContext, stateManager, exitOnNextTick, FeatureFlags, pathManager } from 'amplify-cli-core';
+import { collectDirectivesByType, readProjectConfiguration } from 'graphql-transformer-core';
+import { printer, prompter } from 'amplify-prompts';
+import { DocumentNode, FieldNode, DirectiveNode, parse } from 'graphql';
+import path from 'path';
+import fs from 'fs-extra';
+
+async function setNotificationFlag(projectPath: string, flagName: string, value: boolean): Promise<void> {
+  let config = stateManager.getCLIJSON(projectPath, undefined, {
+    throwIfNotExist: false,
+    preserveComments: true,
+  }) || { features: { graphqltransformer: {} } };
+
+  if (!config.features) {
+    config.features = { graphqltransformer: {} };
+  }
+
+  if (!config.features.graphqltransformer) {
+    config.features.graphqltransformer = {};
+  }
+
+  config.features.graphqltransformer[flagName] = value;
+  stateManager.setCLIJSON(projectPath, config);
+  await FeatureFlags.reloadValues();
+}
+
+export async function notifyFieldAuthSecurityChange(context: $TSContext): Promise<void> {
+  const flagName = 'showfieldauthnotification';
+  const dontShowNotification = !FeatureFlags.getBoolean(`graphqltransformer.${flagName}`);
+
+  if (dontShowNotification) return;
+
+  const projectPath = pathManager.findProjectRoot() ?? process.cwd();
+  const meta = stateManager.getMeta(projectPath);
+
+  const apiNames = Object.entries(meta?.api || {})
+    .filter(([_, apiResource]) => (apiResource as $TSAny).service === 'AppSync')
+    .map(([name]) => name);
+
+  const doesNotHaveGqlApi = apiNames.length < 1;
+
+  if (doesNotHaveGqlApi) {
+    return await setNotificationFlag(projectPath, flagName, false);
+  }
+
+  const apiName = apiNames[0];
+  const apiResourceDir = pathManager.getResourceDirectoryPath(projectPath, 'api', apiName);
+  const project = await readProjectConfiguration(apiResourceDir);
+  const directiveMap = collectDirectivesByType(project.schema);
+  const doc: DocumentNode = parse(project.schema);
+  const fieldDirectives: Set<string> = hasFieldAuthDirectives(doc);
+
+  if (displayAuthNotification(directiveMap, fieldDirectives)) {
+    printer.blankLine();
+    const continueChange = await prompter.yesOrNo(
+      `This version of Amplify CLI introduces additional security enhancements for your GraphQL API. ` +
+        `The changes are applied automatically with this deployment. This change won't impact your client code. Continue`,
+    );
+
+    if (!continueChange) {
+      await context.usageData.emitSuccess();
+      exitOnNextTick(0);
+    }
+
+    const schemaPath = path.join(apiResourceDir, 'schema.graphql');
```

Comment:
this needs to handle the case for projects with `schema` directory

Reply:
a similar change had to be added for the other notification: https://github.com/aws-amplify/amplify-cli/pull/9670

Reply:
thanks for pointing this out - fixed.